File: neural_networks/two_py/src/envs.py
import os

# path to save/load weights file
CACHEPATH = os.getenv('CACHEPATH', './cache.bin')
MAXERAS = os.getenv('MAXERAS', 500)  # max era loops

ALPHA = os.getenv('ALPHA', 0.5)  # learning rate
EPSILON = os.getenv('EPSILON', 0.005)  # lower error bound
DECRAISE = os.getenv('DECRAISE', 0.999)  # alpha decraise

WMSIZE = os.getenv('WMSIZE', 40)
HMSIZE = os.getenv('HMSIZE', 40)
MSIZE = WMSIZE * HMSIZE  # matrix size

H1 = os.getenv('H1', 280)  # count of neurons on first hidden layer
H2 = os.getenv('H2', 120)  # count of neurons on second hidden layer
OL = os.getenv('OL', 10)  # count of neurons on output layer

File: neural_networks/two/src/envs.rs
pub const IMG_FILES_PATH: &str = "./.tmp/images";
pub const SHAPES: [char; 10] = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];

pub const MSIZE: usize = 80*80; // input matrix size

pub const H1: usize = 8; // count of neurons on first layer
pub const H2: usize = 12; // count of neurons on second layer
pub const OL: usize = 10; // count of neurons on output layer

pub const _UPPERB: f32 = 0.8; // upper bound
pub const _LOWERB: f32 = 0.2; // lower bound

static mut _ALPHA: f32 = 0.1; // learning speed
